### PR TITLE
Armondo Doc Controllers

### PR DESCRIPTION
**Pull request by GomezArmondo**
Made documentation for the controllers package and the Cors package
Documentation only no code added/removed

### CODE REVIEW

File: src/main/java/com/cepengagementservice/Config/CorsConfig.java [hunk 1]
```diff
@@ -5,21 +5,24 @@
 import org.springframework.web.servlet.config.annotation.CorsRegistry;
 import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
 
-    @Configuration
-    public class CorsConfig{
+/**
+ * Allows for CORS
```
Comment (AnthonyNGarcia) [inline src/main/java/com/cepengagementservice/Config/CorsConfig.java]: Try to explain in the class JDoc why this class exists, and what functionalities it can offer at a high level.

File: src/main/java/com/cepengagementservice/Config/CorsConfig.java [hunk 2]
```diff
@@ -5,21 +5,24 @@
...
-                    .allowedMethods("GET","POST","PUT", "DELETE")
-                    .allowedHeaders("*")
-                    .allowedOrigins("http://localhost:3000");
-                }
-            };
-        }
-    }
+				registry.addMapping("/**").allowedMethods("GET", "POST", "PUT", "DELETE").allowedHeaders("*")
+						.allowedOrigins("http://localhost:3000");
+			}
```
Comment (AnthonyNGarcia) [inline src/main/java/com/cepengagementservice/Config/CorsConfig.java:33]: I don't see any JDocs for these methods. Add them to all of these methods.

File: pom.xml [hunk 3]
```diff
@@ -68,7 +68,7 @@
 		<dependency>
 			<groupId>com.h2database</groupId>
 			<artifactId>h2</artifactId>
-			<scope>runtime</scope>
+			<!-- <scope>runtime</scope> -->
```
Comment (smgruver) [inline pom.xml:75]: I think this can stay, but could you leave a comment for why it would or wouldn't be commented out?

File: src/main/java/com/cepengagementservice/Controllers/RequestController.java [hunk 4]
```diff
@@ -25,9 +25,15 @@
 	
 	@GetMapping("/interventions")
 	public List<Request> getAllInterventions(){
```
Comment (AnthonyNGarcia) [inline src/main/java/com/cepengagementservice/Controllers/RequestController.java]: This class is missing JDocs

File: src/main/java/com/cepengagementservice/Controllers/UserBatchController.java [hunk 5]
```diff
@@ -22,6 +22,10 @@
     @Autowired
     private UserBatchService userBatchService;
```
Comment (AnthonyNGarcia) [inline src/main/java/com/cepengagementservice/Controllers/UserBatchController.java:33]: Missing Class JDocs

File: pom.xml (hunk 3 above)
Comment (GomezArmondo) [reply pom.xml:75]: The h2-console can not be accesses if the scope is runtime Commenting it would will make it default till console isn't needed

File: src/main/java/com/cepengagementservice/Controllers/UsersControllers.java [hunk 6]
```diff
@@ -20,6 +20,11 @@
 import org.springframework.web.bind.annotation.RestController;
 import com.cepengagementservice.Services.RequestService;
 
+/**
+ * 
+ * @author Unknown
+ *
+ */
```
Comment (AnthonyNGarcia) [inline src/main/java/com/cepengagementservice/Controllers/UsersControllers.java]: Missing populated class JDocs

File: src/main/java/com/cepengagementservice/Config/CorsConfig.java [hunk 7]
```diff
@@ -5,21 +5,24 @@
...
-        @Bean
-        public WebMvcConfigurer corsConfigurer(){
-            return new WebMvcConfigurer(){
+	@Bean
+	public WebMvcConfigurer corsConfigurer() {
+		return new WebMvcConfigurer() {
 
-                @Override
-                public void addCorsMappings(CorsRegistry registry){
+			@Override
```
Comment (smgruver) [inline src/main/java/com/cepengagementservice/Config/CorsConfig.java:28]: This method should have its own JavaDoc, including what it accepts as input and what it does with it.

**Review by AnthonyNGarcia — CHANGES_REQUESTED** [review]

File: src/main/java/com/cepengagementservice/Config/CorsConfig.java [hunk 8]
```diff
@@ -5,21 +5,24 @@
 import org.springframework.web.servlet.config.annotation.CorsRegistry;
 import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
 
-    @Configuration
-    public class CorsConfig{
+/**
+ * Allows for CORS
+ * @author Unknown
+ *
+ */
```
Comment (smgruver) [inline src/main/java/com/cepengagementservice/Config/CorsConfig.java]: The class seems simple but more detail would be appreciated, inlcuding the method it has.

**Comment by AnthonyNGarcia** [comment]
Pull dev-trev and merge into your branch to resolve merge conflicts

File: src/main/java/com/cepengagementservice/Controllers/AssociateGraphController.java [hunk 9]
```diff
@@ -1,70 +1,88 @@
...
-	import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
-	import org.springframework.web.bind.annotation.GetMapping;
-	import org.springframework.web.bind.annotation.PathVariable;
-	import org.springframework.web.bind.annotation.RestController;
-	import org.springframework.web.client.RestTemplate;
+/**
+ * Functions used to retrive Associate's graphing
+ * @author Unknown
+ *
+ */
```
Comment (smgruver) [inline src/main/java/com/cepengagementservice/Controllers/AssociateGraphController.java]: Please include details about what specific methods the class contains

File: src/main/java/com/cepengagementservice/Controllers/RequestController.java [hunk 10]
```diff
@@ -25,9 +25,15 @@
...
+		//Add more validation 
 		return is.findAll();
```
Comment (smgruver) [inline src/main/java/com/cepengagementservice/Controllers/RequestController.java:37]: Missing JDocs for class

File: src/main/java/com/cepengagementservice/Controllers/UserBatchController.java [hunk 11]
```diff
@@ -22,6 +22,10 @@
...
 
+    /**
```
Comment (smgruver) [inline src/main/java/com/cepengagementservice/Controllers/UserBatchController.java:34]: Missing JDocs for class

File: src/main/java/com/cepengagementservice/Controllers/UsersControllers.java (hunk 6 above)
Comment (smgruver) [inline src/main/java/com/cepengagementservice/Controllers/UsersControllers.java]: Empty?

File: src/main/java/com/cepengagementservice/Controllers/UsersControllers.java [hunk 12]
```diff
@@ -41,8 +47,14 @@
 
     // Future, maybe return user.
     // Change logic in service.
+    /**
+     * Add a user
+     * @param User user 
+     * @return ResponseEntity<?>(<?> Extends object)
+     */
```
Comment (smgruver) [inline src/main/java/com/cepengagementservice/Controllers/UsersControllers.java]: Include that the password gets hashed

File: src/main/java/com/cepengagementservice/Services/SNSPublisherService.java [hunk 13]
```diff
@@ -34,6 +34,8 @@
 
 	public void publisher(Request request) {
 		try{
+			System.out.println("In inside SNSPublisher publisher");
+
```
Comment (smgruver) [inline src/main/java/com/cepengagementservice/Services/SNSPublisherService.java]: Remove, please

File: src/main/java/com/cepengagementservice/security/AuthenticationController.java [hunk 14]
```diff
@@ -42,7 +42,7 @@
   @RequestMapping(value = "${jwt.get.token.uri}", method = RequestMethod.POST)  //Gets Login URI from app.properties and Generates Token if valid
   public ResponseEntity<?> createAuthenticationToken(@RequestBody JwtTokenRequest authenticationRequest)
       throws AuthenticationException { // in case the request is not valid
-
+System.out.println("Creat Authentication Token");
```
Comment (smgruver) [inline src/main/java/com/cepengagementservice/security/AuthenticationController.java]: Remove, please

File: src/main/java/com/cepengagementservice/security/AuthenticationController.java [hunk 15]
```diff
@@ -55,6 +55,8 @@
 
   @RequestMapping(value = "${jwt.refresh.token.uri}", method = RequestMethod.GET)  //Gets refresh URI from app.properties and refreshes token if it is expired, needs fix.
   public ResponseEntity<?> refreshAndGetAuthenticationToken(HttpServletRequest request) {
+	  System.out.println("refresh  Authentication Token");
```
Comment (smgruver) [inline src/main/java/com/cepengagementservice/security/AuthenticationController.java]: Remove, please

File: src/main/java/com/cepengagementservice/security/AuthenticationController.java [hunk 16]
```diff
@@ -70,10 +72,14 @@
 
   @ExceptionHandler({ AuthenticationException.class })
   public ResponseEntity<String> handleAuthenticationException(AuthenticationException e) {
+	  System.out.println("handle  Authentication Token");
+
```
Comment (smgruver) [inline src/main/java/com/cepengagementservice/security/AuthenticationController.java]: Remove, please

File: src/main/java/com/cepengagementservice/security/AuthenticationController.java [hunk 17]
```diff
@@ -70,10 +72,14 @@
...
     return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body(e.getMessage());
   }
 
   private void authenticate(String email, String password) {
+	  System.out.println("authenticate");
+
```
Comment (smgruver) [inline src/main/java/com/cepengagementservice/security/AuthenticationController.java]: Remove, please

File: src/main/java/com/cepengagementservice/security/JwtTokenAuthorizationOncePerRequestFilter.java [hunk 18]
```diff
@@ -38,11 +38,12 @@
     @Override
     protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain chain) throws ServletException, IOException {
         logger.debug("Authentication Request For '{}'", request.getRequestURL());
-
+        
         final String requestTokenHeader = request.getHeader(this.tokenHeader); // If request header doesn't match this.tokenHeader it returns null
 
         String email = null; //changed from username
         String jwtToken = null;
+        System.out.println(requestTokenHeader);
```
Comment (smgruver) [inline src/main/java/com/cepengagementservice/security/JwtTokenAuthorizationOncePerRequestFilter.java]: Remove, please

Comment (smgruver) [inline src/main/java/com/cepengagementservice/security/JwtTokenAuthorizationOncePerRequestFilter.java]: Remove, please

File: src/main/java/com/cepengagementservice/security/JwtTokenAuthorizationOncePerRequestFilter.java [hunk 19]
```diff
@@ -53,6 +54,7 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
                 logger.warn("JWT_TOKEN_EXPIRED", e);
             }
         } else {
+        	System.out.println("if the requestTokenHeader does not exist and/or doesn't start with Bearer");
```
Comment (smgruver) [inline src/main/java/com/cepengagementservice/security/JwtTokenAuthorizationOncePerRequestFilter.java]: Remove, please

**Review by smgruver — DISMISSED** [review]
Need more details in some places

File: src/main/java/com/cepengagementservice/Controllers/RequestController.java [hunk 20]
```diff
@@ -22,12 +22,21 @@
 	
 	@Autowired
 	private SNSPublisherService snsPublisherService;
-
```
Comment (AnthonyNGarcia) [inline src/main/java/com/cepengagementservice/Controllers/RequestController.java:25]: Missing Class JDocs

File: src/main/java/com/cepengagementservice/Services/SNSPublisherService.java [hunk 21]
```diff
@@ -34,6 +34,8 @@
 
 	public void publisher(Request request) {
 		try{
+			System.out.println("In inside SNSPublisher publisher");
```
Comment (AnthonyNGarcia) [inline src/main/java/com/cepengagementservice/Services/SNSPublisherService.java]: Missing Class JDoc, publisher method JDoc, and remove the sysout

File: src/main/java/com/cepengagementservice/security/AuthenticationController.java (hunk 14 above)
Comment (AnthonyNGarcia) [inline src/main/java/com/cepengagementservice/security/AuthenticationController.java]: Remove all the sysouts

File: src/main/java/com/cepengagementservice/security/JwtTokenAuthorizationOncePerRequestFilter.java (hunk 18 above)
Comment (AnthonyNGarcia) [inline src/main/java/com/cepengagementservice/security/JwtTokenAuthorizationOncePerRequestFilter.java]: Remove syso

File: src/main/resources/application.properties [hunk 22]
```diff
@@ -13,16 +12,14 @@ spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.PostgreSQLDialect
...
 spring.datasource.url=jdbc:h2:mem:testdb;MODE=PostgreSQL
 spring.datasource.driver-class-name=org.h2.Driver
 spring.datasource.data=classpath:data.sql
-
 spring.h2.console.enabled=true
 
 #changed from sa to user
-spring.datasource.username=sa
+spring.datasource.username=user
 spring.datasource.password=
```
Comment (AnthonyNGarcia) [inline src/main/resources/application.properties]: why did you change the username to user?

**Review by AnthonyNGarcia — CHANGES_REQUESTED** [review]

**Review by AnthonyNGarcia — APPROVED** [review]

**Review by kdgreen398 — APPROVED** [review]
